protocol/shannon/protocol.go

```diff
@@ -55,6 +56,20 @@ type Protocol struct {
 
 	// HTTP client used for sending relay requests to endpoints while also capturing & publishing various debug metrics.
 	httpClient *httpClientWithDebugMetrics
+
+	// fallbackEndpoints contains the fallback endpoint configurations for the protocol.
+	// The fallback endpoints are used when no endpoints are available for
+	// the requested service. For example, if all protocol endpoints are sanctioned,
+	// the fallback endpoints will be used to populate the list of endpoints.
+	// Each service can have a SendAllTraffic flag to control traffic routing behavior.
+	fallbackEndpoints map[protocol.ServiceID]serviceFallbackConfig
+}
+
+// serviceFallbackConfig holds the fallback configuration for a service,
+// including the endpoints and whether to send all traffic to fallback.
+type serviceFallbackConfig struct {
```

Comment:
WHy not reuse the structure in `protocol/shannon/config.go`?

Reply:
They are not the same:

`protocol/shannon/protocol.go`
```go
type serviceFallbackConfig struct {
	SendAllTraffic bool
	Endpoints      map[protocol.EndpointAddr]endpoint
}
```

`protocol/shannon/config.go`
```go
ServiceFallback struct {
		ServiceID    protocol.ServiceID `yaml:"service_id"`
		FallbackURLs []string           `yaml:"fallback_urls"`
		// If true, all traffic will be sent to the fallback endpoints for the service,
		// regardless of the health of the protocol endpoints.
		SendAllTraffic bool `yaml:"send_all_traffic"`
	}
)```

Reply:
IMO protocol is the correct place for it. It's used directly only in the protocol related code and is a property of the `Protocol` struct. It is not a config level struct.